test/unit/content-info.test.js

```diff
@@ -0,0 +1,435 @@
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                defaultExtension: 'json',
+                format: 'json',
+                mimeType: 'text',
+                name: '响应',
+                ext: 'json'
+            });
+    });
+
+    it('Should return file name from content disposition header with encoded type iso-8859-1', function () {
+        var response = new Response({ header: [
+            {
+                key: 'Content-Disposition',
+                value: 'attachment;filename*=iso-8859-1\'\'myResponse.json'
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                defaultExtension: 'json',
+                format: 'json',
+                mimeType: 'text',
+                name: 'myResponse',
+                ext: 'json'
+            });
+    });
+
+    it('filename* parameter is preferred than filename in content disposition header', function () {
+        var response = new Response({ header: [
+            {
+                key: 'Content-Disposition',
+                value: 'attachment;filename*=utf-8\'\'%E5%93%8D%E5%BA%94.json; filename = response.json'
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                defaultExtension: 'json',
+                format: 'json',
+                mimeType: 'text',
+                name: '响应',
+                ext: 'json'
+            });
+    });
+
+    it('Should return file name from content disposition header with type form-data and file name', function () {
+        var response = new Response({ header: [
+            {
+                key: 'Content-Disposition',
+                value: 'form-data; filename="testResponse.json"'
+            },
+            {
+                key: 'Content-Type',
+                value: 'application/json'
+            }
+        ], stream: Buffer.from('a test json').toJSON
+        });
+        expect(contentInfo.contentInfo(response)).to
+            .eql({
+                charset: 'utf8',
+                defaultExtension: 'json',
+                format: 'json',
+                mimeType: 'text',
+                name: 'testResponse',
+                ext: 'json'
+            });
+    });
+    describe('Regex DOS Security', function () {
+        it('fileNameRegex should be thwarted for long patterns of ACII char', function () {
+            this.timeout(1000);
+            var filenameHeader = 'attachment;filename=' + 'hello.txt.'.repeat(8000000),
```

Comment:
@dhineshwiz Could you switch all numbers here and in all ReDos tests below to exponential notation? Makes them a whole lot easier to read 😄